openwisp_notifications/tasks.py

```diff
@@ -25,3 +31,66 @@ def delete_obsolete_notifications(instance_app_label, instance_model, instance_i
         | Q(target_object_id=instance_id)
     )
     Notification.objects.filter(where).delete()
+
+
+# Following tasks add notification settings in database
+
+
+@shared_task
+def ns_register_notification_type(notification_type):
+    """
+    Adds notification setting for all users who are memeber of an
+    organization when a new notification type is registered.
+    """
+    for user in User.objects.all():
+        if not user.organizations_dict.keys():
+            try:
+                NotificationSetting.objects.create(
+                    user_id=user.id, type=notification_type, organization_id=None
+                )
+            except IntegrityError:
+                pass
+            finally:
+                continue
+
+        for org_user in user.organizations_dict:
+            try:
+                NotificationSetting.objects.create(
+                    user_id=user.id, organization_id=org_user, type=notification_type,
+                )
+            except IntegrityError:
+                continue
+
+
+@shared_task
+def ns_unregister_notification_type(notification_type):
```

Comment:
I am not loving these names, what about names like `create_settings_for_` and `delete_settings_for_`?

Reply:
Those were looking long, `ns` is short for `notificationsetting`